Improve DocParse documentation

MarkLindblad opened the pull request.

AbhijitP-009 reviewed (commented):
Can we also fix the docs here: https://docs.aryn.ai/docparse/aryn_sdk#partitioning-asynchronously

and replace them with what's here: https://abaft-koi-a54.notion.site/DocParse-Async-API-188e241bf7cd80fa95e3e8b904cef699?

AbhijitP-009 commented on sdk-reference/partition.mdx:
```diff
@@ -283,21 +357,28 @@ an api key. If aryn_api_key is set it will override this. The default ArynConfig
 "result" which contains what would have been returned had `partition_file` been called directly. "status" can be
 "done", "pending", "error", or "no_such_job".<br />
+If there is an error with partitioning itself rather than scheduling the work, then the "status" will be "done" and the
+"result" will contain an "error" field. Unlike `partition_file`, `partition_file_async_result` does
```
I think we can leave out the last sentence here.

MarkLindblad replied:
Removed.

AbhijitP-009 commented on sdk-reference/partition.mdx:
````diff
@@ -283,21 +357,28 @@ an api key. If aryn_api_key is set it will override this. The default ArynConfig
   "result": ...
 }
 ```
+| `status` | `status_code` | `result` present? |
````
I think the last column can just be `result`. And we can describe what the result is for "done" and for the others we can simply say "no result returned"

MarkLindblad replied:
Done.

MarkLindblad commented:
> Can we also fix the docs here: https://docs.aryn.ai/docparse/aryn_sdk#partitioning-asynchronously
> 
> and replace them with what's here: https://abaft-koi-a54.notion.site/DocParse-Async-API-188e241bf7cd80fa95e3e8b904cef699?

I've now changed the example code to match the notion doc and added the long-form explanation of using async DocParse from `aryn-sdk`.

MarkLindblad commented on docparse/aryn_sdk.mdx:
```diff
@@ -101,7 +101,11 @@ The full chunking options are documented [here](/docparse/chunking_strategies).
 ## Partitioning asynchronously
 
-You can submit a job to the async DocParse API using aryn-sdk and the pattern below:
+The `partition_file_async_submit` function will return immediately after being called with a job id you can use to keep
```
`partition_file_async_submit` is a synchronous function you call so that you can partition documents asynchronously. It's not async. It's blocking when you call it.

AbhijitP-009 replied:
Ok, i see. How about : `partition_file_async_submit` is a function which submits a file partitioning job to Aryn and returns with a job_id.  You can use the job_id to keep track of your request to partition the file. You can poll...

MarkLindblad replied:
I've now rephrased it.

MarkLindblad commented on sdk-reference/partition.mdx:
```diff
@@ -272,32 +346,38 @@ while True:
+| "done" | 200 | matches what would have been returned had `partition_file` been called directly |
+| "pending" | 202 | not present |
+| "no_such_job" | 404 | not present |
+| "error" | all others | not present |
```
If the status code is anything other than 200, 202, or 404, the "status" will be "error" since we don't expect any other response. It's not necessarily 5xx.

MarkLindblad commented on sdk-reference/partition.mdx:
```diff
@@ -272,32 +346,38 @@ while True:
+ssl_verify is `True` by default, enforcing SSL verification.
 </Accordion>
 <Accordion title="Return Value">
 A dict like the below containing "status" and "status_code". When "status" is "done", the returned dict also contains
```
Done.

MarkLindblad commented on sdk-reference/partition.mdx:
```diff
@@ -223,22 +223,96 @@ elts_and_dataframes = tables_to_pandas(data)
-A dict like `{"job_id": "aryn:j-yjchiesfqlish1ye58cco41"}` containing the job_id of the submitted request.
+A dict like `{"job_id": "aryn:j-yjchiesfqlish1ye58cco41"}` containing the `job_id` of the submitted request.
+
+If no Aryn API key is provided, `partition_file_async_submit` will raise an `requests.exceptions.HTTPError`.
```
Done.

MarkLindblad commented on sdk-reference/partition.mdx:
```diff
@@ -223,22 +223,96 @@ elts_and_dataframes = tables_to_pandas(data)
 
 <Accordion title="Return Value">
-A dict like `{"job_id": "aryn:j-yjchiesfqlish1ye58cco41"}` containing the job_id of the submitted request.
+A dict like `{"job_id": "aryn:j-yjchiesfqlish1ye58cco41"}` containing the `job_id` of the submitted request.
```
I've now removed the example.

MarkLindblad commented on sdk-reference/partition.mdx:
```diff
@@ -223,22 +223,96 @@ elts_and_dataframes = tables_to_pandas(data)
+instead include a field called `markdown` that contains a string representing the entire document in Markdown format.
+- `ssl_verify`: A `boolean` that controls whether the client verifies the SSL certificate of the chosen DocParse
+server. ssl_verify is `True` by default, enforcing SSL verification.
+- `aryn_api_key`: An Aryn API key, provided as a string. You can get one for free at
```
Done.

MarkLindblad commented on sdk-reference/partition.mdx:
```diff
@@ -272,32 +346,38 @@ while True:
-- `job_id` Required. str of the job id to cancel
-- `aryn_api_key`: An Aryn API key, provided as a string. You can get one for free at
+- `job_id`: Required. A str of the job id to poll and attempt to get the result for.
+- `aryn_api_key`: A str containing your Aryn API key. You can get one for free at
```
Done.

AbhijitP-009 commented on docparse/aryn_sdk.mdx:
```diff
@@ -101,7 +101,13 @@ The full chunking options are documented [here](/docparse/chunking_strategies).
 ## Partitioning asynchronously
 
-You can submit a job to the async DocParse API using aryn-sdk and the pattern below:
+`partition_file_async_submit` is a function that submits a file partitioning job to Aryn and returns with its `job_id`.
```
Let's change job to task

AbhijitP-009 commented on docparse/aryn_sdk.mdx:
```diff
@@ -143,10 +149,19 @@ results = [None] * len(files)
 for i, job_id in enumerate(job_ids):
```
job to task

AbhijitP-009 commented on sdk-reference/partition.mdx:
```diff
@@ -223,22 +223,96 @@ elts_and_dataframes = tables_to_pandas(data)
+
+
 ## partition_file_async_submit
 Submit a document for asynchronous partitioning and get its `job_id`. Meant to be used with `partition_file_async_result`.
```
job to task

AbhijitP-009 reviewed (commented):
Let's change all mentions of job to task. Let's also wait on this until the code changes are actually ready.

MarkLindblad commented:
> Let's change all mentions of job to task. Let's also wait on this until the code changes are actually ready.

This documentation reflects the currently deployed and released code. I will make a new pr to reflect the upcoming changes.

AbhijitP-009 reviewed (approved):
LGTM for now. We'll have to make more changes as we implement things
